### DISCUSSION TITLE
Getting number as decimal, not as exponential?

### DISCUSSION THREAD

**Discussion by Sabeurre** [ImageMagick/ImageMagick#2466] [Help] [answered]
Hello,
Keep in mind I'm fairly new to imagemagick and scripting.
I made a .bat script which uses the number outputed by imagemagick:
magick %%i -fill White +opaque Black -precision 15 -format %%["fx:(1-mean)*100"] info:"black.txt"
The problem is, sometimes I will get a really low number like 1.11022302462516e-14 which my bat script can't use and treats it as 1.
Is it possible to get the number as decimal, or is it possible to round the number?
IM 7.0.10 on windows

**Comment by fmw42** [comment, marked as answer]
Apart from increasing precision, you could multiply by a large number in your fx expression, then divide by the same amount in your bat script in Windows. Our you can round using `"[fx:round((1-mean)*100)]"` or just test using a ternary expression and report 0 if mean is too close to 1 `"[fx:mean>0.9999?0:(1-mean)*100]"`

Note that your expression is wrong. The quotes need to be outside the brackets as `"[...]"` Also to save to a file, you should redirect using `> black.txt`

**Comment by snibgo** [comment]
The construct ...
```
info:"black.txt"
```
... should work fine, and could be faster than ...
```
info: >black.txt
```
... as IM writes directly to the file, instead of indirectly via shell indirection. On Windows, it is slightly faster, but this is only noticeable for large outputs such as "txt:out.txt".
